### PR TITLE
Add more documentation to extensible classes

### PR DESCRIPTION
**Pull request by noryb009**
This rephrases and extends the current documentation. New documentation is also added, including extensible enums, comparisons to other approaches, and connector guards.

### CODE REVIEW

File: doc/compiler/extensible_classes/Extensible_Classes.md [hunk 1]
```diff
@@ -161,11 +184,45 @@ Pros:
...
+### Namespace Aliases
+
+Namespace aliases are a proposed alternative to connectors.
+This method makes namespaces accumulative, where if a class is not found in a namespace at compile time, the compiler looks in its parent namespace.
+
+This greatly reduces the boilerplate code required to implement connectors, including the macros defined above a file's includes.
+Additionally, this simplifies the question of what to inherit from.
+
+This method however has a few cons.
+If the wrong class is included when inherited from, it would not be used in the class hierarchy.
```
Comment (mgaudet) [inline doc/compiler/extensible_classes/Extensible_Classes.md]: This sentence is a little strange...

**Comment by mgaudet** [comment]
You appear to be working w/ the 1-line per sentence model of prose. While that makes sense for rendered content (preserves diffability), IMO, for our markdown we ought to wrap at 80.

**Comment by noryb009** [comment]
@mgaudet Thanks for reviewing.

There's currently a mix of documentation styles (eg. each paragraph on one line, to 80 characters per line), so we might need to have a wider discussion about what style we should use moving forward. For now, I've changed this patch to the 80 characters per line style.

**Comment by 0xdaryl** [comment]
$0.02.  From a reviewer's perspective, the other advantage of wrapping at 80 chars is that its easier to add per-line comments closer to where the comment actually applies (as opposed to the end of a paragraph).

**Pull request merged**